src/analyses/ai.h

```diff
@@ -408,17 +409,18 @@ class ait:public ai_baset
   }
 
   /// Used internally by the analysis.
-  std::unique_ptr<statet> abstract_state_before(locationt t) const override
+  std::shared_ptr<const statet>
+  abstract_state_before(locationt t) const override
   {
     typename state_mapt::const_iterator it = state_map.find(t);
     if(it == state_map.end())
     {
-      std::unique_ptr<statet> d = util_make_unique<domainT>();
+      auto d = std::make_shared<domainT>();
       CHECK_RETURN(d->is_bottom());
       return d;
     }
 
-    return util_make_unique<domainT>(it->second);
+    return std::make_shared<domainT>(domainT(it->second));
```

Comment:
??? By the end of the PR it is...?  I have tried to break into smaller commits so that this is more reviewable and so that each commit has a single purpose.